experimental/delegate/mlp.mlir

```diff
@@ -0,0 +1,60 @@
+// RUN: iree-compile --iree-preprocessing-transform-spec-filename=%p/mlp_spec.mlir  %s | \
+// RUN: iree-run-module --device=local-sync \
+// RUN:     --executable_plugin=$IREE_BINARY_DIR/runtime/plugins/AMD-AIE-experimental/delegate/mlp_bf16_aie_delegate$IREE_DYLIB_EXT \
+// RUN:     --module=- \
+// RUN:     --function=mlp_invocation \
+// RUN:     --input="256x256xf32=2" \
+// RUN:     --input="256x256xf32=3"
+
+// The implementation of MLP is matched using a transform dialect script and is forwarded to a system plugin.
+
+#x86_64_target = #hal.executable.target<"llvm-cpu", "embedded-elf-x86_64", {
+  data_layout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128",
+  native_vector_size = 32 : index,
+  target_triple = "x86_64-none-elf"
+}>
+
+// The target devices that the program will run on. We can compile and run with
+// multiple targets, but this example is maintaining an implicit requirement
+// that the custom kernel being spliced in is supported by the target device,
+// hence we only support llvm-cpu here.
+#cpu_target = #hal.device.target<"llvm-cpu", [
+  #x86_64_target
+]>
+
+#map = affine_map<(d0, d1) -> (d0, d1)>
+module @example attributes {hal.device.targets = [#cpu_target]} {
+
+  // CHECK-LABEL: EXEC @mlp_invocation
+  //       CHECK: [Plugin]: M = 2, N = 2, K = 2
+  //       CHECK: 2x2xf32=[-12 0][0 -12]
```

Comment:
I am not following how we ended up with M=2 with M=256 in the above command?

Reply:
The test is disabled, as Eric says ctests don't run on the hardware.  If needed, I will remove the test portion.  However, this file also serves as the source of the (manual for now) e2e test

Reply:
yes makes sense to remove the CHECK-LABEL and `RUN` in front of the command.

Reply:
Done